e2e-tests/IIoTPlatform-E2E-Tests/RegistryHelper.cs

```diff
@@ -70,55 +70,125 @@ public class RegistryHelper : IDisposable {
-                        configuration, ct);
-                        return added;
+                        return await RegistryManager.AddConfigurationAsync(configuration, ct).ConfigureAwait(false);
                     }
-                    catch (DeviceAlreadyExistsException) when (forceUpdate) {
-                        //
-                        // Technically update below should now work but for
-                        // some reason it does not.
+                    catch (DeviceAlreadyExistsException) {
+                        // Technically update below should now work but for some reason it does not.
                         // Remove and re-add in case we are forcing updates.
-                        //
                         _context.OutputHelper?.WriteLine("IoT Hub device configuration already existed, remove and recreate it");
-                        await RegistryManager.RemoveConfigurationAsync(configuration.Id, ct);
-                        var added = await RegistryManager.AddConfigurationAsync(
-                            configuration, ct);
-                        return added;
+                        await RegistryManager.RemoveConfigurationAsync(configuration.Id, ct).ConfigureAwait(false);
+                        return await RegistryManager.AddConfigurationAsync(configuration, ct).ConfigureAwait(false);
                     }
                 }
 
-                _context.OutputHelper?.WriteLine("IoT Hub device configuration will be updated");
-                // Try update existing configuration
-                var result = await RegistryManager.UpdateConfigurationAsync(
-                    configuration, forceUpdate, ct);
-                return result;
+                if (Equals(configuration, getConfig)) {
+                    return getConfig;
+                }
+
+                _context.OutputHelper?.WriteLine("Existing IoT Hub device configuration is different, remove and recreate it");
+                await RegistryManager.RemoveConfigurationAsync(configuration.Id, ct).ConfigureAwait(false);
+                return await RegistryManager.AddConfigurationAsync(configuration, ct).ConfigureAwait(false);
             }
             catch (Exception e) {
                 _context.OutputHelper?.WriteLine("Error while creating or updating IoT Hub device configuration! {0}", e.Message);
                 throw;
             }
         }
 
+        /// <summary>
+        /// Check equality of two deployment configurations.
+        /// </summary>
+        public static bool Equals(Configuration c0, Configuration c1) {
+            if (c0.Id != c1.Id) {
+                return false;
+            }
+
+            if (c0.TargetCondition != c1.TargetCondition) {
+                return false;
+            }
+
+            if (c0.Priority != c1.Priority) {
+                return false;
+            }
+
+            if (c0.Content.ModulesContent.Count != c1.Content.ModulesContent.Count) {
```

Comment:
>c0.Content.ModulesContent.Count != c1.Content.ModulesContent.Count [](http://example.com/codeflow?start=16&length=66)

don't we need some null checks for Content and ModulesContent here?

Reply:
Good point. Added.